dsp-shared/src/test/scala/dsp/valueobjects/SchemaSpec.scala

```diff
@@ -0,0 +1,483 @@
+import zio.prelude.Validation
+import zio.test._
+import dsp.constants.SalsahGui
+
+/**
+ * This spec is used to test the [[dsp.valueobjects.User]] value objects creation.
+ */
+object SchemaSpec extends ZIOSpecDefault {
+
+  private val guiElementUnknownIri  = "http://www.knora.org/ontology/salsah-gui#Unknown"
+  private val guiElementListIri     = SalsahGui.List
+  private val guiElementPulldownIri = SalsahGui.Pulldown
+  private val guiElementRadioIri    = SalsahGui.Radio
+  private val guiElementSliderIri   = SalsahGui.Slider
+  private val guiElementCheckboxIri = SalsahGui.Checkbox
+
+  private val guiAttributeInvalidString            = "invalid"
+  private val guiAttributeUnknownString            = "unknown=10"
+  private val guiAttributeSizeString               = "size=80"
+  private val guiAttributeSizeStringKey            = "size"
+  private val guiAttributeSizeStringValue          = "80"
+  private val guiAttributeSizeStringWithWhitespace = "size    =  80 "
+  private val guiAttributeHlistString              = "hlist=http://rdfh.ch/lists/082F/PbRLUy66TsK10qNP1mBwzA"
+  private val guiAttributeMinString                = "min=1"
+  private val guiAttributeMaxString                = "max=10"
+
+  private val guiAttributeSize  = Schema.GuiAttribute.make(guiAttributeSizeString).fold(e => throw e.head, v => v)
+  private val guiAttributeHlist = Schema.GuiAttribute.make(guiAttributeHlistString).fold(e => throw e.head, v => v)
+  private val guiAttributeMin   = Schema.GuiAttribute.make(guiAttributeMinString).fold(e => throw e.head, v => v)
+  private val guiAttributeMax   = Schema.GuiAttribute.make(guiAttributeMaxString).fold(e => throw e.head, v => v)
+
+  private val guiElementList     = Schema.GuiElement.make(guiElementListIri).fold(e => throw e.head, v => v)
+  private val guiElementPulldown = Schema.GuiElement.make(guiElementPulldownIri).fold(e => throw e.head, v => v)
+  private val guiElementRadio    = Schema.GuiElement.make(guiElementRadioIri).fold(e => throw e.head, v => v)
+  private val guiElementSlider   = Schema.GuiElement.make(guiElementSliderIri).fold(e => throw e.head, v => v)
+  private val guiElementCheckbox = Schema.GuiElement.make(guiElementCheckboxIri).fold(e => throw e.head, v => v)
+
+  def spec = (
+    guiAttributeTest +
+      guiElementTest +
+      guiObjectTest +
+      validateGuiObjectsPointingToListTest +
+      validateGuiObjectSliderTest +
+      guiObjectListTest +
+      guiObjectRadioTest +
+      guiObjectPulldownTest +
+      guiObjectSliderTest +
+      guiObjectCheckboxTest
+  )
+
+  private val guiAttributeTest = suite("gui attribute")(
+    test("pass an empty value and return an error") {
+      assertTrue(
+        Schema.GuiAttribute.make("") == Validation.fail(BadRequestException(SchemaErrorMessages.GuiAttributeMissing))
+      )
+    },
+    test("pass an invalid value and return an error") {
+      assertTrue(
+        Schema.GuiAttribute.make(guiAttributeInvalidString) == Validation.fail(
```

Comment:
In my opinion, crating an additional variable instead of adding the string here directly is not more readable. The text in the name of the unit test is more than clear enough what will be passed. If anything, now I need to scroll up all the time to see the string against which the test is run, to check if it is actually correct.

Reply:
Very elegant solution @BalduinLandolt but in this particular case, when string is used only once, wrapping it into val doesn't make sense. Also tests tends to be more dirty in terms of repetitions anyway, just to make things more visible/readable.

Reply:
I followed Ivan's and Marcin's opinion here and replaced the `val`s with the actual strings where it made sense to me. The unit tests here are very simple indeed but for more complex tests I like Balduin's idea of defining given and expected values outside of the asserts. 

I agree with Balduin's suggestion about the Exception and try to change that in the value objects.